### PR TITLE
MS Graph Office365: PAPP-27394 - Bug fix

### PR DESCRIPTION
**Pull request by ighelani-crest**
**NOTE :** 
The static pipeline is failing as the update playbooks are yet not merged in the next branch, once playbooks are merged it will pass.

#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [x] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers), 
  please document them in the `readme.html`.
- [ ] If this is a new connector or you are adding new actions
    - [ ] Please document in the `readme.html` all methods (eg, OAuth) used to authenticate 
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `readme.html`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through 
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

  

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

File: office365.json [hunk 1]
```diff
@@ -15,11 +15,11 @@
         }
     ],
     "license": "Copyright (c) 2017-2022 Splunk Inc.",
-    "app_version": "2.7.1",
+    "app_version": "2.8.0",
     "utctime_updated": "2022-02-04T01:20:46.000000Z",
     "package_name": "phantom_msgraphoffice365",
     "main_module": "office365_connector.py",
-    "min_phantom_version": "5.3.0",
+    "min_phantom_version": "5.3.3",
```
Comment (achandya-crest) [inline office365.json:22]: Update min_phantom_version to 5.3.5

File: release_notes/unreleased.md [hunk 2]
```diff
@@ -1 +1,2 @@
 **Unreleased**
+* Bug fix for the extraction of custom email headers [PAPP-26852]
```
Comment (achandya-crest) [inline release_notes/unreleased.md:2]: Can also mention URL regex change

Comment (ighelani-crest) [reply release_notes/unreleased.md:2]: Done

File: office365.json (hunk 1 above)
Comment (ighelani-crest) [reply office365.json]: Done

**Review by mmomin-crest — APPROVED** [review]

**Comment by hardik-crest** [comment]
> Any update on this pull request? It's been opened for a while, and blocks the 'send email' action

Hello, @alnumac
The development is done and testing in almost completed. Will be releasing this app soon.

File: office365_connector.py [hunk 3]
```diff
@@ -2394,115 +3276,128 @@ def handle_action(self, param):
 
         self.debug_print("action_id", self.get_action_identifier())
 
-        if action_id == 'test_connectivity':
+        if action_id == "test_connectivity":
```
Comment (tonyc-phantom) [inline office365_connector.py:3279]: I'm not sure I understand the point of changing all of the formatting from single to double quotes here: Python standards say to use single quotes unless there is a single quote character in the string itself, which it won't ever be.

File: process_email.py [hunk 4]
```diff
@@ -263,28 +261,31 @@ def _extract_urls_domains(self, file_data, urls, domains, parent_id=None):
...
-
-        uri_text = [uri for uri in uri_text if uri.startswith('http')]
-        uris.extend(uri_text)
-
-        # Parse it as a text file
-        uris_from_text = re.findall(uri_regexc, file_data)
-        if uris_from_text:
-            uris_from_text = [self._clean_url(uri) for uri in uris_from_text]
-        uris.extend(uris_from_text)
+        if links or srcs:
```
Comment (tonyc-phantom) [inline process_email.py:264]: Then used single quotes here? Just wondering why two different styles were used.

File: office365_connector.py (hunk 3 above)
Comment (hardik-crest) [reply office365_connector.py:3279]: Hey @tonyc-phantom 
The changes were made by the external contributor, therefore haven't changed them.

**Review by tonyc-phantom — CHANGES_REQUESTED** [review]
Please upgrade the django package in requirements.txt to 3.2.19 due to this vulnerability: https://nvd.nist.gov/vuln/detail/CVE-2023-31047.

**Review by tonyc-phantom — APPROVED** [review]

**Pull request merged**